### PR TITLE
Fix circle test reporting

### PR DESCRIPTION
**Pull request by jimmay5469**
Resolves: #677 

### Bug Fixes
* Capture and re-throw original exit status so that CircleCI successfully reports test failures
* Fix (or remove) broken tests to get back to green

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -338,8 +338,8 @@ jobs:
               mix coveralls.html --exclude no_geth --parallel --umbrella
             else
               mix coveralls.circle --exclude no_geth --parallel --umbrella ||
-                # if mix failed, then coveralls_merge won't run, so single done here
-                curl -k https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN -d "payload[build_num]=$CIRCLE_WORKFLOW_WORKSPACE_ID&payload[status]=done"
+                # if mix failed, then coveralls_merge won't run, so signal done here and return original exit status
+                (local retval=$? && curl -k https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN -d "payload[build_num]=$CIRCLE_WORKFLOW_WORKSPACE_ID&payload[status]=done" && return $retval)
```
Comment (jimmay5469) [inline .circleci/config.yml]: If there's a better way to do this let me know.

Comment (jimmay5469) [reply .circleci/config.yml]: Also, it seems like `coverage/coveralls` and `ci/circleci: coveralls_merge` are still hanging on this branch, so is the `curl` statement even doing what it's trying to do?

Comment (Lucasnar) [reply .circleci/config.yml]: Shouldn't we use `exit` instead of `return` here? Not sure how this is used by CircleCI, though.

Comment (jimmay5469) [reply .circleci/config.yml]: When I was testing this out locally `exit` was also closing my terminal window so I started using `return` to be able to play around easier. It looks like CircleCI is now picking up the failures with `return` but if `exit` is better I'm all for it.

Comment (Lucasnar) [reply .circleci/config.yml]: I don't think `exit` is necessarily better. If it works with `return` and makes testing easier, I think it is good to keep it. 👍

**Comment by Stamates** [comment]
@igorffs There's some failing tests (locally) remaining... the first 3 are flaky, I don't have enough context on the last to help resolve.  Maybe you know a way to fix these?:
* `test/explorer/smart_contract/verifier_test.exs:15`
* `test/explorer/smart_contract/solidity/code_compiler_test.exs:52`
* `test/explorer/smart_contract/solidity/code_compiler_test.exs:83`
* `test/indexer/block_fetcher/realtime_test.exs:41`
The last one is the only one failing on this CircleCI build

**Comment by KronicDeth** [comment]
@Stamates, @jimmay5469 I removed the `local ` in front of `retval` as the logs were reporting `/bin/bash: line 6: local: can only be used in a function`.

**Comment by KronicDeth** [comment]
https://circleci.com/workflow-run/f501d97c-26c1-4823-bad6-5e8b7400e349 demonstrates that the failures are still reported after the removal of `local` before `retval`.

**Pull request merged**